src/passes/OptimizeInstructions.cpp

```diff
@@ -1259,8 +1277,24 @@ struct OptimizeInstructions
         matches(curr, binary(Abstract::And, pure(&left), ival(0)))) {
       return right;
     }
+    // (signed)x % C_pot == 0   ==>  x & (C_pot - 1) == 0
+    // (signed)x % C_pot != 0   ==>  x & (C_pot - 1) != 0
+    if (matches(curr,
+                binary(Abstract::Eq,
+                       binary(&inner, Abstract::RemS, any(&left), constant(&c)),
+                       ival(0))) ||
+        matches(curr,
+                binary(Abstract::Ne,
+                       binary(&inner, Abstract::RemS, any(&left), constant(&c)),
+                       ival(0)))) {
```

Comment:
I'm wondering is it possible made something like this?
```cpp
    if (matches(curr,
                binary(Abstract::any(Abstract::Eq, Abstract::Ne),
                       binary(&inner, Abstract::RemS, any(&left), constant(&c)),
                       ival(0)))) {
  ....
}
```

Reply:
Now you can extract the `BinaryOp` to examine it separately if you'd like. I agree that when the pattern covers multiple lines, repeating it isn't so nice.